hi-ml/testhiml/testhiml/test_run_ml.py

```diff
@@ -457,12 +495,12 @@ def test_model_weights_when_resume_training() -> None:
     experiment_config = ExperimentConfig(model="HelloWorld")
     container = HelloWorld()
     container.max_num_gpus = 0
-    container.src_checkpoint = CheckpointParser(mock_run_id(id=0))
```

Comment:
I'm not sure this change has the same behaviour, we want to make sure we're able to resume from a previous aml run, str(hello_world_checkpoint) is a local path outside AML

Reply:
I made this change mostly for test speed. I would assume that all the handling of the different checkpoint types is done correctly by the `CheckpointParser` class, so I don't need to check it here again.